akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -228,6 +228,45 @@ class ConnectionPoolSpec extends AkkaSpec("""
       acceptIncomingConnection()
       val (Success(_), 42) = responseOut.expectNext()
     }
+
+    "never close hot connections when minConnections key is given and >0" in new TestSetup() {
```

Comment:
I'm struggling with writing a proper test for `minConnections`. Kind of testability vs encapsulation battle.
After the approach change what this test should do (I think) is reach to `slotStates` and inspect whether enough slots did connect (aren't idle). Ideally we would probably go even deeper - to slots themselves (to validate if conductor statuses reflect real life). 
The problem is that `slotStates` is well hidden - internal variable of a function, inside a private class, used only to build a sub-graph in a bigger one ... I was looking for a way to expose it, but my approach felt ugly and left a trail of `Internal API, only for testing` comments. 
Could anyone point me to an existing piece of code doing something similar?
cc: @ktoso @2beaucoup